### DISCUSSION TITLE
Ability to search through citations

### DISCUSSION THREAD

**Discussion by tokisuno** [zk-org/zk#390] [Ideas]
### Verify

- [X] I searched the existing discussions for similar ideas

### Share your idea or feature request

I write literary notes with zk and nvim, and use a separate extension to cite articles I'm referring to in my notes. Although I do already use tags for broad topics (philo, gender, ling), it would be nice to have some way to search by the pandoc @citation1999 format through my notes.

**Comment by tjex** [comment]
I have it in the back of my mind to extend the telescope functionality for grepping. So we can grep for text through our notes, but have returned in the telescope window the notes title and not the file name (which for many is an alphanumerical id, so not that descriptive). 

In your case I would think it should suffice, as the citation itself is unique.

**Reply by tokisuno** [reply]
I did find a workaround for this. I use telescope-bibtex. Below is my config. I define a custom citation format and force that when I am working on my zettelkasten. 

https://github.com/tokisuno/nvim/blob/main/lua/lucas/lazy/telescope.lua

**Reply by tjex** [reply]
oh nice. I totally overlooked this extension 🫨
This is going to help a lot for my masters thesis!
